File: Day2/soln2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define INPUT_PATH "inputs/Day2.txt"

int main() {
    FILE *input;

    int depth = 0;
    int aim = 0;
    int horizontal = 0;

    input = fopen(INPUT_PATH, "r");

    int x = 0;
    char command[10]; // largest word is `forward`
    fscanf(input, "%s %d", command, &x);
    while(!feof(input)) {
        if(strcmp(command, "forward") == 0) {
            horizontal += x;
            depth += aim * x;
        }
        else if(strcmp(command, "up") == 0) {
            aim -= x;
        }
        else if(strcmp(command, "down") == 0) {
            aim += x;
        }
        else {
            perror("Error: no match");
            return 1;
        }
        fscanf(input, "%s %d", command, &x);
    }

    printf("First: %d\n", aim * horizontal);
    printf("Second: %d\n", depth * horizontal);
}

File: Day2/soln2.py
import os
dirname = os.path.dirname(__file__)
inputPath = os.path.join(dirname, '../inputs/Day2.txt')

with open(inputPath) as file:
    lines = file.readlines()

depth = 0
horizontal = 0
aim = 0

for line in lines:
    arr = line.split()
    if arr[0] == "forward":
        horizontal += int(arr[1])
        depth += aim * int(arr[1])
    if arr[0] == "up":
        aim -= int(arr[1])
    if arr[0] == "down":
        aim += int(arr[1])

print("First: " + str(aim * horizontal))
print("Second: " + str(depth * horizontal))



